src/components/control/date.jsx

```diff
@@ -119,6 +134,15 @@ const Date = forwardRef(function Date(props, forwardedRef) {
     if (active && props.onChange) props.onChange({ target: ref.current });
   }, [value]);
 
+  useEffect(() => {
+    if (!previousActive && active) {
```

Comment:
Hmmm... would using a dependency on the active state work as well? I would prefer to stick to React conventions.

```
useEffect(() => {
  if (active) {
    props.onActive(value);
  } else { 
    props.onDeactive(value);
  }
}, [active]);

Reply:
the problem with this implementation is is we immediately call `props.onDeactive ` when we render the component for the first time. Which I mean it is inactive, but it's not the same as changing from active to inactive, and from inactive to active. 

One option is we could make a hook like this. https://thewebdev.info/2021/03/13/how-to-make-the-react-useeffect-hook-not-run-on-initial-render/ and then we would have the cleaner version of the `useEffect`

Reply:
ah, okay, that works. we also have a `useMountedEffect` which only runs the effect after it has been mounted. looks like both achieve the same thing.